bug: builder phase validation fails with FAILED when PR exists due to API propagation race in retry loop

## Summary

When the builder creates a PR successfully but the GitHub API hasn't propagated it yet, the shepherd's builder phase validation loop incorrectly returns `PhaseStatus.FAILED` even though the work is complete. The shepherd then reverts the issue label and exits early, requiring manual re-runs.

## Observed Behavior

From shepherd log (issue #2922, PR #2954):
```
[OK] Direct completion: PR created for issue #2922
[OK] Direct completion succeeded
[WARN] No PR found for issue #2922 but checkpoint indicates pr_created — retrying after 2s for API propagation
[ERROR] builder phase validation failed: worktree exists (branch=feature/issue-2922, commits_ahead=1, uncommitted=none); remote branch exists; PR #2954 (with loom:review-requested); ...
[INFO] Label cleanup: reverted issue #2922 from loom:building to loom:issue
[INFO] Skipping systematic failure counter: PR #2954 already exists for issue #2922 (builder succeeded)
[OK] Released claim for issue #2922
```

Note the contradiction: the last two lines show the shepherd **knows** the builder succeeded (PR exists), yet still returned `FAILED` and released the claim.

## Root Cause

The bug is in the `while True` validation loop in `BuilderPhase.run()` (`loom-tools/src/loom_tools/shepherd/phases/builder.py`):

```python
while True:
    if self.validate(ctx, quiet=True):
        break  # Validation passed

    diag = self._gather_diagnostics(ctx)

    if not self._has_incomplete_work(diag):
        # ... falls through to FAILED path
        return PhaseResult(status=PhaseStatus.FAILED, message=f"builder phase validation failed: {diag['summary']}")
```

The sequence that triggers the bug:

1. `direct_completion()` creates the PR and returns `True` → loop calls `continue`
2. Loop re-enters: `self.validate()` calls `validate_phase()` which calls `_find_pr_for_issue()`
3. `_find_pr_for_issue()` returns `None` (API propagation delay — 2s retry inside `validate_phase` wasn't sufficient)
4. `validate_phase()` returns `satisfied=False`
5. Loop falls to `diag = self._gather_diagnostics(ctx)` — this call finds the PR (different API call, slightly later, propagation complete)
6. `_has_incomplete_work(diag)` checks: `commits_ahead=1` AND `pr_number=2954` AND `pr_has_review_label=True` → returns `False` (workflow done per [line 3905](https://github.com/rjwalters/loom/blob/main/loom-tools/src/loom_tools/shepherd/phases/builder.py#L3905))
7. Since `_has_incomplete_work` is False AND `_is_no_changes_needed` is False → **falls through to FAILED**

The missing check: when `_has_incomplete_work` returns `False` because the PR exists with the correct label (not because there's nothing to do), the loop should treat this as **success**, not failure.

## Expected Behavior

When `_has_incomplete_work(diag)` returns `False` because `diag["pr_number"] is not None and diag["pr_has_review_label"]` is `True`, the validation loop should return `PhaseStatus.SUCCESS` (workflow complete), not `PhaseStatus.FAILED`.

## Proposed Fix

In the `not _has_incomplete_work` branch, add a check before falling through to FAILED:

```python
if not self._has_incomplete_work(diag):
    # NEW: check if workflow is already complete (PR with correct label)
    if (
        diag.get("pr_number") is not None
        and diag.get("pr_has_review_label", False)
    ):
        # Workflow is done — validate_phase just had an API propagation delay
        return PhaseResult(
            status=PhaseStatus.SUCCESS,
            message=f"PR #{diag['pr_number']} exists with loom:review-requested",
            phase_name="builder",
        )
    
    # existing no-changes-needed check
    if self._is_no_changes_needed(diag):
        ...
    
    # existing FAILED fallthrough
    ...
```

## Reproduction

This race condition is most likely to trigger immediately after `_direct_completion()` creates a PR — the 2-second retry inside `validate_phase` (#2710) is insufficient when the GitHub API propagation takes longer.

Observed consistently across multiple shepherd runs for issue #2922 during the session on 2026-02-19.

## Impact

- Shepherd exits after builder phase despite successful PR creation
- Issue label reverted from `loom:building` to `loom:issue`
- Requires manual re-run of shepherd to proceed to judge phase
- Wastes ~5-7 minutes per occurrence (builder re-run + worktree cleanup)
- Systematic failure counter is correctly skipped (shepherd detects PR exists), but the label revert still triggers confusion

## Related

- Issue #2710 (added the 2s API propagation retry inside `validate_phase`)
- `loom-tools/src/loom_tools/shepherd/phases/builder.py` lines ~865-910 (validation loop)
- `loom-tools/src/loom_tools/shepherd/phases/builder.py` lines ~3886-3945 (`_has_incomplete_work`)

--- a/loom-tools/src/loom_tools/shepherd/phases/builder.py
+++ b/loom-tools/src/loom_tools/shepherd/phases/builder.py
@@ -990,6 +990,30 @@ class BuilderPhase:
 
             # Check if this is incomplete work that could be completed
             if not self._has_incomplete_work(diag):
+                # Check if workflow is already complete (PR with correct label).
+                # This catches the API propagation race where validate_phase()
+                # returns False (PR not yet visible) but _gather_diagnostics()
+                # finds it moments later after the API propagates.
+                # See issue #2961.
+                if diag.get("pr_number") is not None and diag.get(
+                    "pr_has_review_label", False
+                ):
+                    pr = diag["pr_number"]
+                    ctx.pr_number = pr
+                    ctx.report_milestone("pr_created", pr_number=pr)
+                    return PhaseResult(
+                        status=PhaseStatus.SUCCESS,
+                        message=(
+                            f"builder phase complete - PR #{pr} exists with "
+                            f"loom:review-requested (API propagation delay resolved)"
+                        ),
+                        phase_name="builder",
+                        data={
+                            "pr_number": pr,
+                            "api_propagation_race": True,
+                        },
+                    )
+
                 # Check if this is the "no changes needed" pattern
                 if self._is_no_changes_needed(diag):
                     log_info(

